eta/classifiers/tfslim_classifiers.py

```diff
@@ -250,50 +253,124 @@ def features_dim(self):
 
         return dim
 
+    @property
+    def exposes_probabilities(self):
+        '''Whether this classifier exposes probabilities for predictions.'''
+        return True
+
+    @property
+    def num_classes(self):
+        '''The number of classes for the model.'''
+        return self._num_classes
+
+    @property
+    def class_labels(self):
+        '''The list of class labels generated by the classifier.'''
+        return self._class_labels
+
     def get_features(self):
-        '''Gets the features generated by the classifier from its last call to
-        `predict()`.
+        '''Gets the features generated by the classifier from its last
+        prediction.
 
         Returns:
-            the feature vector, or None if the classifier has not (or cannot)
-                generate features
+            an array of features, or None if the classifier has not (or does
+                not) generate features
         '''
-        if not self.generates_features:
+        if not self.exposes_features:
             return None
 
         return self._last_features
 
+    def get_probabilities(self):
```

Comment:
I like it as implemented because it gives implementations the flexibility to store their probabilities however they choose. Maybe they choose to access them directly from a network graph, for example. Who knows

It is true that all of the implementations currently in our library follow a similar pattern of using `self._last_probs`, but it need not be the case